tests/test_instantiate.py

```diff
@@ -8,13 +8,22 @@
 import torch
 from torch import Tensor
 from torch import nn
-model = nn.Linear(1, 2)
+model = nn.Linear(1, 1)
 
 @pytest.mark.parametrize(
     "classname, cfg, passthrough_kwargs, expected",
     [
+        pytest.param("Adadelta", {"lr": 0.1}, {"params":model.parameters()}, Adadelta(lr=0.1, params=model.parameters()), id="AdadeltaConf"),
+        pytest.param("Adagrad", {"lr": 0.1}, {"params":model.parameters()}, Adagrad(lr=0.1, params=model.parameters()), id="AdagradConf"),
         pytest.param("Adam", {"lr": 0.1}, {"params":model.parameters()}, Adam(lr=0.1, params=model.parameters()), id="AdamConf"),
+        pytest.param("Adamax", {"lr": 0.1}, {"params":model.parameters()}, Adamax(lr=0.1, params=model.parameters()), id="AdamaxConf"),
         pytest.param("AdamW", {"lr": 0.1}, {"params":model.parameters()}, AdamW(lr=0.1, params=model.parameters()), id="AdamWConf"),
+        pytest.param("ASGD", {"lr": 0.1}, {"params":model.parameters()}, ASGD(lr=0.1, params=model.parameters()), id="ASGD"),
+        pytest.param("LBFGS", {"lr": 0.1}, {"params":model.parameters()}, LBFGS(lr=0.1, params=model.parameters()), id="LBFGS"),
+        pytest.param("RMSprop", {"lr": 0.1}, {"params":model.parameters()}, RMSprop(lr=0.1, params=model.parameters()), id="RMSprop"),
+        pytest.param("Rprop", {"lr": 0.1}, {"params":model.parameters()}, Rprop(lr=0.1, params=model.parameters()), id="Rprop"),
+        pytest.param("SGD", {"lr": 0.1}, {"params":model.parameters()}, SGD(lr=0.1, params=model.parameters()), id="SGD"),
+        pytest.param("SparseAdam", {"lr": 0.1}, {"params":model.parameters()}, SparseAdam(lr=0.1, params=model.parameters()), id="SparseAdam"),
```

Comment:
we need to add formatting with black, isort etc.

Reply:
Looking at the whole unit test, just wondering why do we need this simple nn model.

What do we want this test to do? Do we want to make sure that all optimizers were properly parametrized, right? By that I understand: fields in a given structured config are equal to the arguments of a given optimizer init.

Plus bonus test: default values - if provided, must be equal.

Do I get it right?

Reply:
btw. 🥇 for using pytest parametrization :)

Reply:
> fields in a given structured config are equal to the arguments of a given optimizer init.

Not exactly, OmegaConf supports tuples by converting them to ListConfig.
Even if the default values were lists, a list is not exactly the same as List Config.

An even bigger problem is that as far as I can tell there is no defined protocol for how the init parameters are stored.
The initial attempt was to compare the optimizers, but they do not seem to be implementing a meaningful `__eq__()`.
You are very welcome to suggest an alternative implementation if you think you achieve good testing in a simpler way.

re: default values:
This in general is covered by the tests of instantiate and configen. You can add a few token tests if you want but it may be pretty exhausting to cover everything.

Reply:
Ideally I wanted these tests to do 3 things:
1. Ensure the config exists.
2. Be a valid input to instantiate (and subsequently get an object back).
3. Show that this object works as expected. 

For an optimizer, taking a step seems to prove that it is functioning correctly. Comparing the output of your cfg optimizer and a directly called optimizer is kind of a bonus in that if it didn't work, it doesn't mean we didn't instantiate an optimizer correctly given the config. It just means the optimizer has nondeterministic behavior.